server/app/scanpy_engine/scanpy_engine.py

```diff
@@ -41,6 +42,7 @@ def cells(self):
     def genes(self):
         return self.data.var.index.tolist()
 
+    # Can't seem to cache a view of a dataframe, need to investigate why
```

Comment:
what happens in this scenario?

Reply:
```
Traceback (most recent call last):
  File "/Users/charlotteweaver/Documents/Git/cellxgene/zzz/lib/python3.6/site-packages/Flask-0.12.4-py3.6.egg/flask/app.py", line 1997, in __call__
    return self.wsgi_app(environ, start_response)
  File "/Users/charlotteweaver/Documents/Git/cellxgene/zzz/lib/python3.6/site-packages/Flask-0.12.4-py3.6.egg/flask/app.py", line 1985, in wsgi_app
    response = self.handle_exception(e)
  File "/Users/charlotteweaver/Documents/Git/cellxgene/zzz/lib/python3.6/site-packages/Flask_RESTful-0.3.6-py3.6.egg/flask_restful/__init__.py", line 273, in error_router
    return original_handler(e)
  File "/Users/charlotteweaver/Documents/Git/cellxgene/zzz/lib/python3.6/site-packages/Flask_RESTful-0.3.6-py3.6.egg/flask_restful/__init__.py", line 273, in error_router
    return original_handler(e)
  File "/Users/charlotteweaver/Documents/Git/cellxgene/zzz/lib/python3.6/site-packages/Flask_Cors-3.0.6-py3.6.egg/flask_cors/extension.py", line 161, in wrapped_function
    return cors_after_request(app.make_response(f(*args, **kwargs)))
  File "/Users/charlotteweaver/Documents/Git/cellxgene/zzz/lib/python3.6/site-packages/Flask-0.12.4-py3.6.egg/flask/app.py", line 1540, in handle_exception
    reraise(exc_type, exc_value, tb)
  File "/Users/charlotteweaver/Documents/Git/cellxgene/zzz/lib/python3.6/site-packages/Flask-0.12.4-py3.6.egg/flask/_compat.py", line 32, in reraise
    raise value.with_traceback(tb)
  File "/Users/charlotteweaver/Documents/Git/cellxgene/zzz/lib/python3.6/site-packages/Flask-0.12.4-py3.6.egg/flask/app.py", line 1982, in wsgi_app
    response = self.full_dispatch_request()
  File "/Users/charlotteweaver/Documents/Git/cellxgene/zzz/lib/python3.6/site-packages/Flask-0.12.4-py3.6.egg/flask/app.py", line 1614, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/Users/charlotteweaver/Documents/Git/cellxgene/zzz/lib/python3.6/site-packages/Flask_RESTful-0.3.6-py3.6.egg/flask_restful/__init__.py", line 273, in error_router
    return original_handler(e)
  File "/Users/charlotteweaver/Documents/Git/cellxgene/zzz/lib/python3.6/site-packages/Flask_RESTful-0.3.6-py3.6.egg/flask_restful/__init__.py", line 273, in error_router
    return original_handler(e)
  File "/Users/charlotteweaver/Documents/Git/cellxgene/zzz/lib/python3.6/site-packages/Flask_Cors-3.0.6-py3.6.egg/flask_cors/extension.py", line 161, in wrapped_function
    return cors_after_request(app.make_response(f(*args, **kwargs)))
  File "/Users/charlotteweaver/Documents/Git/cellxgene/zzz/lib/python3.6/site-packages/Flask-0.12.4-py3.6.egg/flask/app.py", line 1517, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "/Users/charlotteweaver/Documents/Git/cellxgene/zzz/lib/python3.6/site-packages/Flask-0.12.4-py3.6.egg/flask/_compat.py", line 32, in reraise
    raise value.with_traceback(tb)
  File "/Users/charlotteweaver/Documents/Git/cellxgene/zzz/lib/python3.6/site-packages/Flask-0.12.4-py3.6.egg/flask/app.py", line 1612, in full_dispatch_request
    rv = self.dispatch_request()
  File "/Users/charlotteweaver/Documents/Git/cellxgene/zzz/lib/python3.6/site-packages/Flask-0.12.4-py3.6.egg/flask/app.py", line 1598, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "/Users/charlotteweaver/Documents/Git/cellxgene/zzz/lib/python3.6/site-packages/Flask_RESTful-0.3.6-py3.6.egg/flask_restful/__init__.py", line 480, in wrapper
    resp = resource(*args, **kwargs)
  File "/Users/charlotteweaver/Documents/Git/cellxgene/zzz/lib/python3.6/site-packages/flask_restful_swagger_2-0.35-py3.6.egg/flask_restful_swagger_2/__init__.py", line 39, in decorator
    return f(*args, **kwargs)
  File "/Users/charlotteweaver/Documents/Git/cellxgene/zzz/lib/python3.6/site-packages/Flask-0.12.4-py3.6.egg/flask/views.py", line 84, in view
    return self.dispatch_request(*args, **kwargs)
  File "/Users/charlotteweaver/Documents/Git/cellxgene/zzz/lib/python3.6/site-packages/Flask_RESTful-0.3.6-py3.6.egg/flask_restful/__init__.py", line 595, in dispatch_request
    resp = meth(*args, **kwargs)
  File "/Users/charlotteweaver/Documents/Git/cellxgene/zzz/lib/python3.6/site-packages/flask_restful_swagger_2-0.35-py3.6.egg/flask_restful_swagger_2/swagger.py", line 219, in inner
    return f(self, *args, **kwargs)
  File "/Users/charlotteweaver/Documents/Git/cellxgene/server/app/rest_api/rest.py", line 203, in get
    filtered_data = data.filter_cells(cells_filter)
  File "/Users/charlotteweaver/Documents/Git/cellxgene/zzz/lib/python3.6/site-packages/flask_cache/__init__.py", line 529, in decorated_function
    rv = self.cache.get(cache_key)
  File "/Users/charlotteweaver/Documents/Git/cellxgene/zzz/lib/python3.6/site-packages/Werkzeug-0.14.1-py3.6.egg/werkzeug/contrib/cache.py", line 312, in get
    return pickle.loads(value)
  File "/Users/charlotteweaver/Documents/Git/cellxgene/zzz/lib/python3.6/site-packages/anndata-0.6.1-py3.6.egg/anndata/base.py", line 408, in __setitem__
    if self._view_args is None:
AttributeError: 'DictView' object has no attribute '_view_args'
```

Reply:
interesting.  looks like anndata DictViews do not support pickling.  https://github.com/theislab/anndata/blob/master/anndata/base.py#L430

Reply:
Oh, thanks for looking that up. Saved me some googling for tomorrow